go/vt/vtgate/vindexes/lookup_internal.go

```diff
@@ -181,6 +190,15 @@ func (lkp *lookupInternal) Delete(vcursor VCursor, rowsColValues [][]sqltypes.Va
 	if lkp.Autocommit {
 		return nil
 	}
+	if len(rowsColValues) == 0 {
+		// This code is unreachable. It's just a failsafe.
+		return nil
+	}
+	// We only need to check the first row. Number of cols per row
+	// is guaranteed by the engine to be uniform.
+	if len(rowsColValues[0]) != len(lkp.FromColumns) {
```

Comment:
Not related to this PR, but there is something I wanted to talk to you about this constraint. We ran into a problem where a multi column vindex has some columns that are nullable and are not provided in the insert. Do you think it would be possible to relax this constraint and maybe only require the leading `from` column and the `to` column ?

Reply:
Yeah. I postponed a feature I was going to add: change v3 code to pass through nulls, and add an option an `allow_nulls` option to lookups.

So, the answer is yes. It's a good idea :)